pipes/WDL/workflows/scaffold_and_refine_multitaxa.wdl

```diff
@@ -0,0 +1,112 @@
+
+    scatter(taxon in taxid_to_ref_accessions) {
+        call ncbi.download_annotations {
+            input:
+                accessions = taxon.right,
+                combined_out_prefix = taxon.left
+        }
+        call assembly.scaffold {
+            input:
+                reads_bam = reads_unmapped_bam,
+                reference_genome_fasta = [download_annotations.combined_fasta],
+                min_length_fraction = 0,
+                min_unambig = 0,
+                allow_incomplete_output = true
+        }
+        call assemble_refbased.assemble_refbased as refine {
+            input:
+                reads_unmapped_bams = [reads_unmapped_bam],
+                reference_fasta     = scaffold.scaffold_fasta,
+                sample_name         = sample_id
+        }
+        # to do: if pre-impute unambig length > some fraction of ref genome, run ncbi.rename_fasta_header and ncbi.align_and_annot_transfer_single
+        # to do: if biosample attributes file provided, run ncbi.biosample_to_genbank
+
+        Map[String, String] stats_by_taxon = {
+            "sample_id" : sample_id,
+            "taxid" : taxon.left,
+
+            "assembly_fasta" : refine.assembly_fasta,
+            "aligned_only_reads_bam" : refine.align_to_self_merged_aligned_only_bam,
+            "coverage_plot" : refine.align_to_self_merged_coverage_plot,
+            "assembly_length" : refine.assembly_length,
+            "assembly_length_unambiguous" : refine.assembly_length_unambiguous,
+            "reads_aligned" : refine.align_to_self_merged_reads_aligned,
+            "mean_coverage" : refine.align_to_self_merged_mean_coverage,
+
+            "intermediate_gapfill_fasta" : scaffold.intermediate_gapfill_fasta,
+            "assembly_preimpute_length_unambiguous" : scaffold.assembly_preimpute_length_unambiguous,
+
+            "replicate_concordant_sites" : refine.replicate_concordant_sites,
+            "replicate_discordant_snps" : refine.replicate_discordant_snps,
+            "replicate_discordant_indels" : refine.replicate_discordant_indels,
+            "replicate_discordant_vcf" : refine.replicate_discordant_vcf,
+
+            "isnvsFile" : refine.align_to_self_isnvs_vcf,
+            "aligned_bam" : refine.align_to_self_merged_aligned_only_bam,
+            "coverage_tsv" : refine.align_to_self_merged_coverage_tsv,
+            "read_pairs_aligned" : refine.align_to_self_merged_read_pairs_aligned,
+            "bases_aligned" : refine.align_to_self_merged_bases_aligned
+        }
+
+        scatter(h in assembly_header) {
```

Comment:
Is the gathered scatter output guaranteed to return in the same order as the input for WDL 1.0 under the different execution engines?
The WDL 1.1 spec [says order is maintained](https://github.com/openwdl/wdl/blob/wdl-1.1/SPEC.md#scatter), but WDL 1.0 [doesn't seem to require same-order output](https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#scatter).

Reply:
I think in the spec it's not guaranteed, but I think all we can say so far is "works as expected in both Terra and miniWDL". I'm not terribly proud of this scatter, and there are actually much [better mechanisms in WDL 1.1](https://github.com/openwdl/wdl/blob/wdl-1.1/SPEC.md#map-functions) that would obviate all this but are not yet available to us in Terra/Cromwell to my knowledge.